src/Mod/Part/App/TopoShapeMapper.cpp

```diff
@@ -0,0 +1,75 @@
+#include "TopoShapeMapper.h"
+
+namespace Part
+{
+
+void ShapeMapper::expand(const TopoDS_Shape &d, std::vector<TopoDS_Shape> &shapes)
+{
+    if (d.IsNull()) return;
+    for(TopExp_Explorer xp(d, TopAbs_FACE);xp.More();xp.Next())
+        shapes.push_back(xp.Current());
+    for(TopExp_Explorer xp(d, TopAbs_EDGE, TopAbs_FACE);xp.More();xp.Next())
+        shapes.push_back(xp.Current());
+    for(TopExp_Explorer xp(d, TopAbs_VERTEX, TopAbs_EDGE);xp.More();xp.Next())
+        shapes.push_back(xp.Current());
+}
+
+void ShapeMapper::populate(bool generated,
```

Comment:
IMO all of these `generated` booleans should be replaced by something along the lines of
```
enum class MappingStatusOrSomething {
    GENERATED,
    MODIFIED
};
```

Reply:
Oh, I guess in modern-ese those shouldn't be all-caps, huh? Old dog, new tricks, etc. `Generated` and `Modified` then, or some such.